code/game/machinery/turret_control.dm

```diff
@@ -165,6 +177,7 @@
 			check_anomalies = value
 
 		updateTurrets()
+		update_icon()
 		return 1
 
 /obj/machinery/turretid/proc/updateTurrets()
```

Comment:
Since `getState` is a thing, change line 184 to be `var/datum/turret_checks/TC = getState()`

Reply:
Added, plus deleted setting TC, since `getState()` does it